src/hooks/useTodo.tsx

```diff
@@ -1,20 +1,30 @@
 import { useEffect, useState } from "react";
 import axios from "axios";
 
-const useTodo = (id) => {
-  const [data, setData] = useState();
+interface Todo {
+  id: number;
+  title: string;
+}
+
+const useTodo = (id: string | undefined) => {
+  const [data, setData] = useState<Todo | undefined>(undefined);
```

Comment:
useState 의 generic 은 생각보다 똑똑해서, 아래처럼 해도 정상작동합니다
```suggestion
  const [data, setData] = useState<Todo>();
```
이게 명시적인 undefined 를 사용하지 않는 더 좋은 방법이기도 하구요

Reply:
오호 굳이 undefined를 명시해주지 않아도 알아듣네요!!! generic..undefined... 메모메모 📝